baseplate/events/queue.py

```diff
@@ -80,6 +80,25 @@ def set_field(self, key, value, obfuscate=False):
             obfuscated_payload = self.payload.setdefault("obfuscated_data", {})
             obfuscated_payload[key] = value
 
+    def unset_field(self, key):
+        """Remove the key from payload if it exists.
+        :param key:
+        :return: value of key that was removed.  None otherwise
+        """
+        return self.payload.pop(key, None)
+
+    def unset_fields(self, keys):
```

Comment:
I'm not really sure we need this function, if a caller wants to do this they can loop themselves.